### PR TITLE
Add support for S3-compatible services

### PR DESCRIPTION
**Pull request by jdeluyck**
This PR adds support for alternative s3 endpoints, which in turn gives support for S3-compatible services.

I'm using this for backing up to Backblaze B2.

### CODE REVIEW

**Comment by jdeluyck** [comment]
Tested default S3 configuration too, which works fine without specifying the endpoints. For alternatives like B2 you need to specify the right region and the right endpoint.

File: README.md [hunk 1]
```diff
@@ -68,6 +68,8 @@ jobs can be set in one file.
    ZFS filesystem.
 #### region : str, default: us-east-1
    S3 region.
+#### endpoint : str, optional
+   S3 endpoint - for S3 compatible services
```
Comment (ddebeau) [inline README.md:72]: Nit: S3 endpoint. Required for S3 compatible services.

**Review by ddebeau — CHANGES_REQUESTED** [review]
Thank you for the PR! I have a couple minor requests.

Tests are failing due to a known GitHub Actions secret limitation:
https://github.community/t/make-secrets-available-to-builds-of-forks/16166/36

File: zfs_uploader/job.py [hunk 2]
```diff
@@ -120,18 +121,21 @@ def __init__(self, bucket_name, access_key, secret_key, filesystem,
             Maximum number of incremental backups per full backup.
         storage_class : str, default: STANDARD
             S3 storage class.
+        endpoint : str
```
[the remark this answers is not in the document]
Comment (jdeluyck) [reply zfs_uploader/job.py]: Adjusted

**Review by ddebeau — APPROVED** [review]

**Pull request merged**